app/assets/javascripts/transitive_dependencies.js

```diff
@@ -0,0 +1,71 @@
+$(document).on('click', '.caret', function () {
+  var current = $(this)
+  var gem_id = $(this).attr('data-gem_id');
+  var version_id = $(this).attr('data-version');
+  $.ajax({
+    type: "get",
+    url: "/gems/"+gem_id+"/versions/"+version_id+"/dependencies.json",
+    success: function(resp) {
+      renderDependencies(resp, gem_id, current)
+    },
+    error: function() {
+      var error_message = "<ul class='deps_item--error'>Request failed. please reload the page and try again</ul>"
+      current.parent().next().next().html(error_message)
+    }
+  });
+})
+
+function renderDependencies(resp, gem_id, current) {
+  scope_display(current, gem_id, resp.run_deps, "runtime")
+  scope_display(current, gem_id, resp.dev_deps, "development")
+  arrow_toggler(current)
+}
+
+function arrow_toggler(current) {
+  var toggler = "<span class='deps_expanded deps_expanded-down'></span>";
+
+  current.parent().click(function() {
```

Comment:
`$(document).on('click', '.scope'` on line 77 registers the function for all elements, you don't need to do this again here.

Reply:
This is for gem_name toggling. Used to toggle hiding both of the scopes and its dependencies. 

.scope toggling is to toggle showing and hiding only the dependencies of that particular scope.

Reply:
.scope will toggle its class scope--expanded and its sibling deps_scope.
![Screenshot from 2019-08-25 11-17-59](https://user-images.githubusercontent.com/16337625/63646096-3566e380-c72a-11e9-9cc1-51c260e4d041.png)

Reply:
While the arrow toggle should toggle it's parents class and its parents sibling's div classes.
The two div classes are used for each scope.
![Screenshot from 2019-08-25 11-21-30](https://user-images.githubusercontent.com/16337625/63646117-b7efa300-c72a-11e9-8f07-a7ee3defa49e.png)

![Screenshot from 2019-08-25 11-23-56](https://user-images.githubusercontent.com/16337625/63646122-deadd980-c72a-11e9-9a43-ef58886ce574.png)